### DISCUSSION TITLE
longhorn-manager sends traffic to non-existent pods Overlay addresses over TCP 8500.

### DISCUSSION THREAD

**Discussion by IntellySenpai** [longhorn/longhorn#5336] [Q&A]
Hello everyone.

I have an issue with one K8s cluster managed in Rancher
It uses CNI with Overlay networking with Calico and Flannel 
And CSI using Longhorn installed from Rancher's plugins market.
My architecture contains 4 workers nodes. On each node there is one longhorn-manager

My node's IP addresses belong to 192.168.0.0/24 with default gateway 192.168.0.1

                                                              W1              W2           W3             W4
My pod's overlay pool belongs to 10.0.1.0/24, 10.0.2.0/24, 10.0.3.0/24, 10.0.4.0/24 ( one to each Worker node )

The issue I am experiencing is that all 4  longhorn-managers are sending continuously traffic with destination IP addresses from the Overlay pool that are not yet allocated

The traffic beahvior is the same on each Worker node
Let's say we pick longhorn-manager on W3:
- The traffic will originate from W3 overlay pool which is specifically assigned to longhorn-manager pod ( 10.0.3.100 )
- The traffic destination will belong to same W3 pool ( 10.0.3.144 ), but is not assigned to any pod on W3 node. 
- Because there is no route to it  ( Calico creates automatically routes towards local pods Overlay address) ( And no destination route over the Flannel tunnel ) the W3 will send the traffic through Default route, meaning towards 192.168.0.1 using the physical interface of W3 node. 
- Because there is no NAT rule for that destination address and because traffic is sent using the physical link the traffic is not VXLAN encapsulated and ends up in the overlay network.

All the traffic generated by longhorn-managers are with Destination TCP port  8500. 

I am not sure how to fix it. How to make Longhorn forget those addresses or make Longhorn remove them and not send request to them ( as I said the addresses do not belong to any pod, so it's basically spoofing... )

My guess is that some pods changed their addresses ( after a failover for example) but Longhorn is not aware of it and still sends traffic to those pods' Overlay addresses. I want to get rid of this leaking of overlay as its generates a lot bogus traffic.

Any suggestions, ideas? Please! anything would be helpful!!!!

**Comment by KillJohn** [comment]
Hello,
I face the same issue. Did you find fix for this issue or workaround?

**Reply by IntellySenpai** [reply]
Hey!
Actually I could not resolve this issue in the long term. 
I restarted at first longhorn managers one by one - and it ended up solving the issue. No more traffic was leaking out.

It worked alright until we updated/patched the worker nodes itself. During the patching we restarted the worker nodes and the next thing I observed was that the issue started again. 1-2 packets per second at the beginning and then to like 4-5 packets per second after 1-3 days.

My team ended up updating the Longhorn Manager to the latest version that was available in Rancher. The update process went fine but the traffic leaking started again soon after the update process.

In my plans are to restart the longhorn managers one more time, one by one, as I did the previous time, and keep them "spoofing" free for as long as possible. I am not sure what other settings I can check to make a workaround or to correct the issue, but as long as the restarts does not affect anything, I think it will be the way to go for now.

Please write me if you come with anything about it.
Cheers!

**Comment by derekbit** [comment]
@weizhe0422 
IIRC, we've raised and discussed the issue before. Do you remember any ideas or suggestions for this issue?

**Comment by veinito** [comment]
We're facing the very same issue...

**Comment by JakubPiasekAGH** [comment]
Infrastructure team bombard us with emails about this isssue :/

**Comment by innobead** [comment]
cc @shuo-wu @PhanLe1010 @c3y1huang
